source/documentBase.py

```diff
@@ -100,13 +110,14 @@ def _maybeGetLayoutTableIds(self, info: textInfos.TextInfo):
 						layoutIDs.add(tableID)
 		return layoutIDs
 
-	def _getTableCellCoords(self, info):
+	def _getTableCellCoords(
+			self,
+			info: textInfos.TextInfo,
+	) -> _TableCell:

```

Comment:
I'm slightly worried about changing return type of this method. While it is private and theoretically should not be called by anything outside `documentBase` this is clearly not the case and some add-on may rely on its result being a tuple and use it as an iterable or rely on the fact that it supports indexing operator. Have you considered using `typing.NamedTuple` - that should give you both type hints and ability to use attribute names rather than rely on indexes, and at the same time preserves backwards compatibility?

Reply:
A line has to be drawn somewhere. A Dataclass is much clearer, and understood by code quality tools better than a NamedTuple. Add-ons shouldn't be relying on underscore methods or classes. If required for an add-on, and issue should be opened.